src/sentry_session.c

```diff
@@ -51,15 +51,21 @@ sentry__session_new(void)
     if (!opts) {
         return NULL;
     }
-    const char *release = sentry_options_get_release(opts);
-    const char *environment = sentry_options_get_environment(opts);
+    char *release = sentry__string_clone(sentry_options_get_release(opts));
     if (!release) {
         return NULL;
     }
 
     sentry_session_t *rv = SENTRY_MAKE(sentry_session_t);
+    if (!rv) {
+        return NULL;
+    }
+
+    char *environment
+        = sentry__string_clone(sentry_options_get_environment(opts));
+
     rv->release = release;
-    rv->environment = environment;
+    rv->environment = sentry__string_clone(environment);
```

Comment:
You're cloning twice here, is this intended?

Reply:
lol, my bad